src/option.h

```diff
@@ -586,6 +586,36 @@ namespace librealsense
        float                   _manual_value;
    };
 
+   /** \brief read_only_option class will permit the user to 
+   *  perform only read (query) of the read_only option when its affecting_option is set */
+   class read_only_option : public proxy_option
```

Comment:
Pls rename class to gated_option as it does both read and write

Reply:
Done